### PR TITLE
refactor(ES6): upgrade BasicEvaluatedExpression to ES6

### PR DESCRIPTION
**Pull request by willmendesneto**
**What kind of change does this PR introduce?**

ES5 => ES6 refactor

**Did you add tests for your changes?**

refactor, all existing tests pass

**If relevant, link to documentation update:**

N/A

**Summary**

Helping in part of the ES6 refactor

**Does this PR introduce a breaking change?**

Nope

### CODE REVIEW

**Comment by willmendesneto** [comment]
@sokra @TheLarkInn Can you guys take a look in this PR, please?

File: lib/BasicEvaluatedExpression.js [hunk 1]
```diff
@@ -2,156 +2,190 @@
...
+		if(quasis === null)
+			delete this.quasis;
+		else
+			this.quasis = quasis;
+		return this;
+	}
+
+	addOptions(options) {
+		if(!this.options) this.options = [];
+		options.forEach(function(item) {
```
Comment (chicoxyzzy) [inline lib/BasicEvaluatedExpression.js]: you can use arrow function here

File: lib/BasicEvaluatedExpression.js [hunk 2]
```diff
@@ -2,156 +2,190 @@
...
-BasicEvaluatedExpression.prototype.setTemplateString = function(quasis) {
-	if(quasis === null)
-		delete this.quasis;
-	else
-		this.quasis = quasis;
-	return this;
+		return undefined;
+	}
+
+	set(value) {
```
Comment (sokra) [inline lib/BasicEvaluatedExpression.js]: Could you check if this method is used?

I think it's unused and we can remove it...

Comment (willmendesneto) [inline lib/BasicEvaluatedExpression.js]: Yes, you're right. This method is not used.

Interesting that we don't have any validation for this inside the repository. I will add these validations in another PR as well.

File: lib/BasicEvaluatedExpression.js (hunk 1 above)
Comment (willmendesneto) [inline lib/BasicEvaluatedExpression.js]: Thanks for the feedback. Done

File: lib/BasicEvaluatedExpression.js [hunk 3]
```diff
@@ -2,156 +2,180 @@
...
+		else if(this.isNumber()) return !!this.number;
+		else if(this.isRegExp()) return true;
+		else if(this.isArray()) return true;
+		else if(this.isConstArray()) return true;
+		else if(this.isWrapped()) return this.prefix && this.prefix.asBool() || this.postfix && this.postfix.asBool() ? true : undefined;
+		else if(this.isTemplateString()) {
+			if(this.quasis.length === 1) return this.quasis[0].asBool();
+			for(var i = 0; i < this.quasis.length; i++) {
+				if(this.quasis[i].asBool()) return true;
+			}
```
Comment (Kovensky) [inline lib/BasicEvaluatedExpression.js]: `if (this.quasis.some(quasis => quasis.asBool())) return true;`

Comment (Kovensky) [inline lib/BasicEvaluatedExpression.js]: Actually, the logic here is wrong; ``` `${0}` ``` should be truthy. But that's a different problem, and a bug in the original 😅

Comment (TheLarkInn) [inline lib/BasicEvaluatedExpression.js]: That's used in a lot of places im surprised we haven't come across that.

Comment (TheLarkInn) [inline lib/BasicEvaluatedExpression.js]: That's used in a lot of places im surprised we haven't come across that.

**Comment by TheLarkInn** [comment]
Nice work thank you very much!

**Pull request merged**